packages/xml-views-quick-fix/src/quick-fix-id.ts

```diff
@@ -0,0 +1,107 @@
+import { find, some } from "lodash";
+import { astPositionAtOffset } from "@xml-tools/ast-position";
+import {
+  XMLDocument,
+  accept,
+  XMLAstVisitor,
+  XMLAttribute,
+  XMLElement,
+} from "@xml-tools/ast";
+import { OffsetRange } from "@ui5-language-assistant/logic-utils";
+
+const ID_PATTERN = /^_IDGen(.+)([1-9]\d*)$/;
+const ID_PREFIX_PATTERN = "_IDGen";
+
+export type QuickFixStableIdInfo = {
+  suggestion: string;
+  offsetRange: OffsetRange;
+};
+
+export function computeQuickFixStableIdInfo(
+  xmlDoc: XMLDocument,
+  errorOffset: OffsetRange
+): QuickFixStableIdInfo | undefined {
+  const biggestIdsCollector = new IdsByElementNameCollectorVisitor();
+  accept(xmlDoc, biggestIdsCollector);
+  const biggestIdsOfElements = biggestIdsCollector.biggestIdsOfElements;
+  const astNode = astPositionAtOffset(xmlDoc, errorOffset.start);
+  if (astNode === undefined || astNode.kind !== "XMLElementOpenName") {
+    return undefined;
+  }
+
+  const xmlElement = astNode.astNode;
```

Comment:
discussion: 
if we know that for `XMLEelementOpenName` there would always be:
- .syntax.openName
- .name

Should we create a branch in the code that is never reached due to type system limitations? or should we define / cast the xmlElement's type more precisely?

 
```typescript
const xmlElement = astNode.astNode as XMLElement & { ... }
```